src/components/Modals/NicknameModal/NicknameModal.tsx

```diff
@@ -32,7 +32,7 @@ const NicknameModal = ({ isModalOpen, onRequestClose }: TNicknameModalProps) =>
 
     const history = useHistory();
     const { error: createError, isError, isSuccess, mutate, reset } = api.advertiser.useCreate();
-    const { setHasCreatedAdvertiser } = useAdvertiserInfoState();
+    const setHasCreatedAdvertiser = useStore(state => state.setHasCreatedAdvertiser);
```

Comment:
Here instead of destructuring the values from the store, we can assign it to a const value here. Reason why I did this is because if I had just destructured it, but without the useShallow as shown below, it would subscribe to the whole store, so any changes to other variables inside the store, it will cause a rerender here in NicknameModal

```
const { setHasCreatedAdvertiser } = useStore();
```

of course we could write it with useShallow, but it is a little bit much to write for one value we need from the store:

```
const { setHasCreatedAdvertiser } = useStore(
        useShallow(state => ({
            setHasCreatedAdvertiser: state.setHasCreatedAdvertiser,
        }))
    );
```

However writing it with useShallow will be much easier to test as when I don't destructure the store values, I have to mock a bit more (check out the BuySellHeader.spec file vs NicknameModal.spec file'